das/worker.go

```diff
@@ -0,0 +1,93 @@
+package das
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"go.uber.org/multierr"
+)
+
+type worker struct {
+	lock     sync.Mutex
```

Comment:
you can just remove the `lock` alias and keep sync.Mutex directly on worker so you can do `w.Lock` instead of `w.lock.Lock`

Reply:
nit but looks nicer imo

Reply:
I usually prefer having it named so mutex public API doesn't accidentally get exported as a struct method.